torchgeo/datasets/south_africa_crop_type.py

```diff
@@ -62,10 +62,9 @@ class SouthAfricaCropType(RasterDataset):
 
     filename_regex = r"""
         ^(?P<field_id>[0-9]*)
-        _(?P<date>[0-9]{4}_[0-9]{2}_[0-9]{2})
+        _(?P<timestamp>[0-9]{4}_07_[0-9]{2})
```

Comment:
Have you confirmed that all locations have a July image? Also, what is the purpose of avoiding the `<date>` name? Although we don't currently support time-series, we plan to in the near future.

Reply:
Wrapping most of your comments together, I have found the bug around a design issue with the dataset as it pertains to the dataloader. Since time series datasets are not currently supported by torchgeo and each field has multiple dates of imagery available, the call to split the dataset into val, test, train splits includes uneven distributions of the original dataset.

For example: In the test data we have two fields 12 & 66 each with a set of imagery from July and another from May. The dataset split with a certain seed includes only may imagery in the test which then contains no valid chips. I can get this to work by changing the regex to only pick up files with the dates we're looking so we are guaranteed valid imagery is in each split but this has limitations as we will still include multiple dates in July for a single field when we only use one which will slow down training. I think a better solution would be to sample the datasets by field ID though I am not sure how to do this. Additionally since S1 and S2 data are in different folders, the R-tree creates an entry for both S1 and S2 for each field leading to similar problems with split. I think this functionality would either have to be added to the base torchgeo-geodataset or this dataset will only work with either s1 or s2 but not both at the same time. This should be fine for our research purposes though.

Reply:
> Additionally since S1 and S2 data are in different folders, the R-tree creates an entry for both S1 and S2 for each field leading to similar problems with split. I think this functionality would either have to be added to the base torchgeo-geodataset or this dataset will only work with either s1 or s2 but not both at the same time.

The "right" way to handle the case of both S1 and S2 is to use `IntersectionDataset` like I did in #1972. But like you said, for our research, we only currently care about S2. So we can leave support for both up to future developers.